keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlResourceRepository.kt

```diff
@@ -242,18 +244,20 @@ open class SqlResourceRepository(
           .orderBy(EVENT.TIMESTAMP.desc())
           .limit(1)
           .fetchOne(EVENT.JSON)
-
-        if (event.javaClass == previousEvent?.javaClass) return@transaction
       }
 
-      txn
-        .insertInto(EVENT)
-        .set(EVENT.UID, ULID().nextULID(event.timestamp.toEpochMilli()))
-        .set(EVENT.SCOPE, event.scope)
-        .set(EVENT.REF, ref)
-        .set(EVENT.TIMESTAMP, event.timestamp)
-        .set(EVENT.JSON, event)
-        .execute()
+      if (event.javaClass == previousEvent?.javaClass) return
+
+      sqlRetry.withRetry(WRITE) {
+        jooq
+          .insertInto(EVENT)
```

Comment:
It looks like the behavior of this block has changed if `event.ignoreRepeatedInHistory` is false. 

Old code: when condition is false, code jumps to line 249, inserts the event

New code: when condition is false, code jumps to line 261, does not insert the event.

Reply:
Ahhh! You're right! What was I thinking! Good spot! 🙏

Reply:
Actually, the qualified was returning to the end of the scope of `jooq.transaction`, which is the last curly brace in the function body before the end of the function, so it was essentially returning from the function altogether if the condition met. Stop scaring me, @lorin! 🤣

Reply:
P.S. But my curly braces were messed up, so thanks! 😄